lms/djangoapps/mobile_api/testutils.py

```diff
@@ -123,8 +141,197 @@ def test_other_user(self):
+        'developer_message':
+            'Cannot access content with unfulfilled pre-requisites or unpassed entrance exam.'
+    }
+
+    # Enrollment list hides enrolled courses, milestone courses are not hidden
+    ALLOW_ACCESS_TO_MILESTONE_COURSE = False  # pylint: disable=invalid-name
+
+    def _add_entrance_exam(self):
+        """ Sets up entrance exam """
+        # Set up the extrance exam
+        self.course.entrance_exam_enabled = True
+
+        self.entrance_exam = ItemFactory.create(  # pylint: disable=attribute-defined-outside-init
+            parent=self.course,
+            category="chapter",
+            display_name="Entrance Exam Chapter",
+            is_entrance_exam=True,
+            in_entrance_exam=True
+        )
+        self.problem_1 = ItemFactory.create(  # pylint: disable=attribute-defined-outside-init
+            parent=self.entrance_exam,
+            category='problem',
+            display_name="The Only Exam Problem",
+            graded=True,
+            in_entrance_exam=True
+        )
+
+        namespace_choices = get_namespace_choices()
+        milestone_namespace = generate_milestone_namespace(
+            namespace_choices.get('ENTRANCE_EXAM'),
+            self.course.id
+        )
+        milestone = {  # pylint: disable=attribute-defined-outside-init
+            'name': 'Test Milestone',
+            'namespace': milestone_namespace,
+            'description': 'Entrance Exam for TestMobileAPIMilestones',
+        }
+        milestone_relationship_types = get_milestone_relationship_types()
+        milestone = add_milestone(milestone)  # pylint: disable=attribute-defined-outside-init
+
+        self.course.entrance_exam_minimum_score_pct = 0.50
+        self.course.entrance_exam_id = unicode(self.entrance_exam.location)
+        modulestore().update_item(self.course, self.user.id)
+
+        # Add the exam
+        add_course_milestone(
+            unicode(self.course.id),
+            milestone_relationship_types['REQUIRES'],
+            milestone
+        )
+        add_course_content_milestone(
+            unicode(self.course.id),
+            unicode(self.entrance_exam.location),
+            milestone_relationship_types['FULFILLS'],
+            milestone
+        )
+
+    def _add_prerequisite_course(self):
+        """ Helper method to set up the prerequisite course """
+        self.prereq_course = CourseFactory.create()  # pylint: disable=attribute-defined-outside-init
+        add_prerequisite_course(self.course.id, self.prereq_course.id)
+
+    def _pass_entrance_exam(self):
+        """ Helper function to pass the entrance exam """
+        # set up the request for exam functions
+        request = RequestFactory()  # pylint: disable=attribute-defined-outside-init
+        request.user = self.user
+        request.COOKIES = {}
+        request.META = {}
+        request.is_secure = lambda: True
+        request.get_host = lambda: "edx.org"
+        request.method = 'GET'
```

Comment:
If this initialization of the request object is needed, there should be common code for it.